### PR TITLE
[doc] Fix closure in manual optimization

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
In most cases, `zero_grad` can be called outside the closure, but `zero_grad` needs to be called inside the closure when using an optimizer like `LBFGS` which requires reevaluation of the loss in `optimizer.step(closure)`.

This PR fixes the closure in the docs so that it works in all cases above.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by rohitgr7 — COMMENTED** [review]
quick question:
is zero_grad and its corresponding hooks called before backward and optimizer.step in case of automatic_optimization??

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
> quick question:
> is zero_grad and its corresponding hooks called before backward and optimizer.step in case of automatic_optimization??

Yes: #6147

**Comment by akihironitta** [comment]
@rohitgr7 Yes, but internally, we call those methods inside `optimizer.step`:
```
optimizer.step(closure=closure)
  closure()
    forward()
    zero_grad()
    backward()
  update_params()
```
but this is equivalent to:
```
forward()
zero_grad()
backward()
optimizer.step(closure=None)
  update_params()
```
All the difference is whether `forward()`, `zero_grad()`, and `backward()` are called inside or outside the closure. We do that because some optimizers, such as LBFGS, need to call them inside the closure. https://pytorch.org/docs/master/optim.html#torch.optim.LBFGS
